drivers/storage/portworx/util/util.go

```diff
@@ -962,6 +970,16 @@ func IsTelemetryEnabled(spec corev1.StorageClusterSpec) bool {
 		spec.Monitoring.Telemetry.Enabled
 }
 
+// IsCCMGoSupported returns true if px version is higher than 2.12
+func IsCCMGoSupported(pxVersion *version.Version) bool {
+	return pxVersion.GreaterThanOrEqual(MinimumPxVersionCCMGO)
+}
+
+// RunCCMGo returns whether to run new ccm go or old ccm java based on if the telemetry proxy image is set
+func RunCCMGo(cluster *corev1.StorageCluster) bool {
+	return cluster.Status.DesiredImages.TelemetryProxy != ""
+}
+
```

Comment:
Don't think this util function correctly conveys the intent. Just checking the status.desiredImages does not tell if CCMGo should be run or not. Does this even need to be a util that anyone can consume? Can it be local to the caller?